### PR TITLE
Added demonstration for network discovery

### PR DESCRIPTION
**Pull request by austinTalbot7241993**
Same thing, new demo requesting feedback.

### CODE REVIEW

**Comment by akotlar** [comment]
Will take a look in the morning

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+    "from sklearn.decomposition import PCA\n",
+    "from sklearn.linear_model import LogisticRegressionCV\n",
+    "from sklearn.metrics import roc_auc_score,roc_curve\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "import torch\n",
+    "\n",
+    "import sys\n",
+    "sys.path.append('/home/austin/Forks/bystro/python/python/bystro/supervised_ppca/')\n",
```
Comment (akotlar) [inline python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:37]: remove

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+    "from gf_dropout_pt import PPCADropout"
```
Comment (akotlar) [inline python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:38]: Should be `from bystro.supervised_ppca.gf_dropout_pt import PPCADropout`

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+    "# Load Dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "myDict = pickle.load(open('/media/austin/ThickBoy__1/DataTST/TST_Data.p','rb'))\n",
```
Comment (akotlar) [inline python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:69]: Can't hardcode the path; please share this file and I will put it in a public repository

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "ename": "NameError",
+     "evalue": "name 'roc_auc' is not defined",
```
Comment (akotlar) [inline python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:218]: Please re-run this cell; this error suggest the cell output is not up to date

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/austin/anaconda3/envs/bystro/lib/python3.10/site-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
```
Comment (akotlar) [inline python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:329]: This error is worrisome; we didn't get convergence, but are showing the results. Can we get convergence?

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb (hunk 3 above)
Comment (akotlar) [reply python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:69]: I see that you provided where to download the files; please update the cell to add a comment that this should be downloaded from https://research.repository.duke.edu/downloads/kd17ct787?locale=en

**Review by akotlar — APPROVED** [review]
LGTM

File: python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb (hunk 5 above)
Comment (austinTalbot7241993) [reply python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb:329]: This method can be slow going to full convergence and this is just an initialization step so we just want to be roughly in the right region.

**Pull request merged**